app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGImageSender.java

```diff
@@ -0,0 +1,78 @@
+package com.teamagam.gimelgimel.app.network.services;
+
+import android.net.Uri;
+import android.util.Log;
+
+import com.teamagam.gimelgimel.app.model.ViewsModels.Message;
+import com.teamagam.gimelgimel.app.model.ViewsModels.MessageImage;
+import com.teamagam.gimelgimel.app.model.entities.ImageMetadata;
+import com.teamagam.gimelgimel.app.network.rest.GGMessagingAPI;
+import com.teamagam.gimelgimel.app.network.rest.RestAPI;
+import com.teamagam.gimelgimel.app.utils.NetworkUtil;
+import com.teamagam.gimelgimel.app.view.viewer.data.geometries.PointGeometry;
+
+import java.io.File;
+
+import okhttp3.MediaType;
+import okhttp3.MultipartBody;
+import okhttp3.RequestBody;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Created on 5/22/2016.
+ * TODO: complete text
+ */
+public class GGImageSender implements IImageSender {
+
+    private static final String LOG_TAG = GGImageSender.class.getSimpleName();
+    private static final String IMAGE_KEY = "image";
+    private static final String IMAGE_MIME_TYPE = "image/*";
+
+    @Override
+    public void sendImage(Uri imageUri, final PointGeometry loc, final long imageTime) {
```

Comment:
we changed it to service so there is no callback availble from the fragment /activity.